libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -131,6 +131,25 @@ namespace eosio { namespace hotstuff {
       warn_hs_message = std::move(warning_hs_message);
    }
 
+   // signed_block is being produced and needs a QC to be put in it before it is broadcast
+   quorum_certificate_message chain_pacemaker::on_block_proposal(const chain::block_id_type& block_id) {
```

Comment:
The name `on_block_proposal` implies on receiving a block proposal. Can it be renamed?

Reply:
`on_producing_block`?